forms/TreeDropdownField.php

```diff
@@ -303,7 +310,8 @@ public function tree(SS_HTTPRequest $request) {
 		if ($this->filterCallback || $this->search != "" )
 			$obj->setMarkingFilterFunction(array($this, "filterMarking"));
 
-		$obj->markPartialTree($nodeCountThreshold = 30, $context = null,
+		$nodeCountThreshold = Config::inst()->get('TreeDropdownField', 'node_threshold_total');
```

Comment:
$this->config() is prefferred to config::inst()->get(), if you are making the other change I requested at the same time. :)